pkg/core/native/native_test/policy_test.go

```diff
@@ -363,3 +368,67 @@ func TestPolicy_BlockedAccounts(t *testing.T) {
 		helperInvoker.Invoke(t, true, "do")
 	})
 }
+
+func TestPolicy_GetBlockedAccounts(t *testing.T) {
+	bc, acc := chain.NewSingleWithCustomConfig(t, func(c *config.Blockchain) {
+		c.Hardforks = map[string]uint32{
+			config.HFFaun.String(): 2,
+		}
+	})
+	e := neotest.NewExecutor(t, bc, acc, acc)
+	p := e.CommitteeInvoker(nativehashes.PolicyContract)
+
+	p.InvokeFail(t, "method not found: getBlockedAccounts/0", "getBlockedAccounts")
+
+	blockedCount := 3
+	blocked := make([]util.Uint160, 0, blockedCount)
+	expected := make([]stackitem.Item, 0, blockedCount)
+	for i := range blockedCount {
+		blocked = append(blocked, e.NewAccount(t, 0).ScriptHash())
+		p.Invoke(t, true, "blockAccount", blocked[i])
+		expected = append(expected, stackitem.Make(blocked[i]))
+	}
+	slices.SortFunc(expected, func(a, b stackitem.Item) int {
+		return bytes.Compare(a.Value().([]byte), b.Value().([]byte))
+	})
+
+	checkGetBlockedAccounts := func(t *testing.T, expected []stackitem.Item) {
+		for i := range expected {
+			w := io.NewBufBinWriter()
+			emit.AppCall(w.BinWriter, p.Hash, "getBlockedAccounts", callflag.All)
+
+			for range i + 1 {
+				emit.Opcodes(w.BinWriter, opcode.DUP)
+				emit.Syscall(w.BinWriter, interopnames.SystemIteratorNext)
+				emit.Opcodes(w.BinWriter, opcode.DROP)
+			}
+
+			emit.Syscall(w.BinWriter, interopnames.SystemIteratorValue)
+			require.NoError(t, w.Err)
+			h := p.InvokeScript(t, w.Bytes(), p.Signers)
+
+			if i < len(expected) {
+				e.CheckHalt(t, h, expected[i])
+			} else {
+				e.CheckFault(t, h, "iterator index out of range")
+			}
+			w.Reset()
+		}
+	}
+
+	checkGetBlockedAccounts(t, expected)
+
+	for unblockedIndex := range blockedCount {
+		p.Invoke(t, true, "unblockAccount", expected[unblockedIndex])
+		updatedBlockAccounts := make([]stackitem.Item, 0, blockedCount-1)
+		updatedBlockAccounts = append(updatedBlockAccounts, expected[:unblockedIndex]...)
+		updatedBlockAccounts = append(updatedBlockAccounts, expected[unblockedIndex+1:]...)
+		checkGetBlockedAccounts(t, updatedBlockAccounts)
+		p.Invoke(t, true, "blockAccount", expected[unblockedIndex])
+	}
```

Comment:
Cycle is not needed, it complicates test. Unblock single account and check, it's enough.